packages/frontend/src/common/components/navigation/Header.tsx

```diff
@@ -1,9 +1,42 @@
-import React from 'react'
+'use client'
+import React, { useEffect, useState } from 'react'
 import Link from 'next/link'
 import Image from 'next/image'
 import { NavigationItem } from '@/common/components/navigation/NavigationItem'
+import { getUser, logOut } from '@/api'
+import { usePathname } from 'next/navigation'
+import { useRouter } from 'next/navigation'
+
+interface Props {
+  id?: string
+  message?: string
+}
 
 export function Header() {
+  const pathname = usePathname()
+  const router = useRouter()
+  const [user, setUser] = useState<Props>({})
+  const protectedPages = ['/submissions', '/invitations', '/invitations/new']
+
+  useEffect(() => {
+    async function getUserData() {
+      const userData = await getUser()
+      setUser(userData)
+    }
+
+    getUserData()
+  }, [pathname])
+
+  function redirectIfPageIsProtected() {
+    if (protectedPages.includes(pathname)) {
+      router.push('/')
+    }
+  }
+
+  function isPageProtected(url: string) {
```

Comment:
This function name lies a bit, or it doesn't tell the whole truth, at least. :)

If I look at `isPageProtected('/submissions')` then I assume it's protected, because `/submissions` is in `protectedPages` array.

Also, is there any difference between `pathname` and `url`?

Reply:
I wonder if it wouldn't be better to have just define function `logOutUser` that would do all the logic 

logOut
setUser
redirectIfPageIsProtected

then we would remove `redirectIfPageIsProtected` and just call the `logOutUser` function from the component below

Reply:
> Also, is there any difference between pathname and url?
Yes, it should be `pathname`